File: problems-extend/3201-3300/3263_convert-doubly-linked-list-to-array-i/provided/javascript/doubly_list_node.js
// Problem-provided doubly linked node (LC 3263 contract).
// The judge's decoder chains next and then wires prev backwards.
class DoublyListNode {
    constructor(val, next, prev) {
        this.val = val ?? 0;
        this.next = next ?? null;
        this.prev = prev ?? null;
    }
}

File: problems-extend-adapt/3201-3300/3263_wiring-a-doubly-list-from-a-chain/provided/rust/doubly_list_node.rs
// Problem-provided doubly linked node (LC 3263 contract).
// Paths are fully qualified: every assembled source shares one module.
pub struct DoublyListNode {
    pub val: i32,
    pub next: Option<std::rc::Rc<std::cell::RefCell<DoublyListNode>>>,
    pub prev: Option<std::rc::Rc<std::cell::RefCell<DoublyListNode>>>,
}

impl DoublyListNode {
    pub fn new(val: i32) -> Self {
        DoublyListNode {
            val,
            next: None,
            prev: None,
        }
    }
}
